exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/ThreadLocalTransactionManager.kt

```diff
@@ -60,15 +66,23 @@ class ThreadLocalTransactionManager(
 
     private class ThreadLocalTransaction(
         override val db: Database,
+        private val setupTxConnection: ((ExposedConnection<*>, TransactionInterface) -> Unit)?,
         override val transactionIsolation: Int,
+        override val readOnly: Boolean,
         val threadLocal: ThreadLocal<Transaction>,
         override val outerTransaction: Transaction?
     ) : TransactionInterface {
 
         private val connectionLazy = lazy(LazyThreadSafetyMode.NONE) {
             outerTransaction?.connection ?: db.connector().apply {
-                autoCommit = false
-                transactionIsolation = this@ThreadLocalTransaction.transactionIsolation
+                setupTxConnection?.invoke(this, this@ThreadLocalTransaction) ?: run {
```

Comment:
What is the idea /use-case of that `setupTxConnection`? Is it expected that all 3 fields bellow can stay unset if `setupTxConnection` was declared?

Reply:
While `ExposedConnection.connection` is of type of `Any` it could be (and actually quite often is) force-casted to a specific implementation which in turn allows to setup vendor specific options. 
This is what it is for.

Reply:
Still don't get it, why not to use `Database.connect(setupConnection = { }) for this?

Reply:
This is a day by day usage. The server connection stays the same for PG, schemas (search path ) is different. Mysql will even re-establish connection I think. 
```
when (dbType) {
    DbType.MYSQL -> {
        connection.schema = schemaName
        connection.catalog = schemaName
    }
    DbType.POSTGRESQL -> {
        connection.schema = schemaName
    }
}
```